R/pdf_array_job.R

```diff
@@ -16,19 +15,29 @@
 #' @param errors Path to write cluster error messages to. Defaults to "/share/temp/sgeoutput/USERNAME/errors"
 #' @param remove_temp_files Remove temporary files after jobs have completed? Defaults to TRUE.
 pdf_array_job <- function(jobs_file, r_script, final_out_dir, 
-                          temp_out_dir=NULL, write_file_name="all_jobs.pdf",
+                          write_file_name="all_jobs.pdf",
                           rshell="/share/singularity-images/lbd/shells/singR.sh -e s", 
                           fthread=5, fmem='10G', h_rt="00:02:00:00", queue="all.q",
                           project="proj_covid", job_name="pdf_array", output=NULL, errors=NULL, 
                           remove_temp_files=TRUE){
-  # Set up output/error logging
+  # Global variables 
   user = Sys.info()[['user']]
+  uuid = uuid::UUIDgenerate()
+  
+  # Set up output/error logging
   if (is.null(output)) output=paste0("/share/temp/sgeoutput/", user, "/output")
+  output = paste0(output, "/", uuid)
+  dir.create(output, recursive=TRUE)
   if (is.null(errors)) errors=paste0("/share/temp/sgeoutput/", user, "/errors")
+  errors = paste0(errors, "/", uuid)
+  dir.create(errors, recursive=TRUE)
```

Comment:
Do we want to put in a if(!dir.exists... here? uuid directory is already created if output same directory as errors

Reply:
Good call - changing now.